### ISSUE TITLE
OVERRIDE_VIDEO_OUTPUT_DIR path is not resolved properly when I use ${PWD} on Windows 10

### ISSUE THREAD

**Issue by fescobar** [aerokube/selenoid#539]
Hi
I have this possible issue.

**Components:**
- Windows 10
- Docker version 18.06.1-ce, build e68fc7a
- docker-compose version 1.22.0, build f46880fe
- Git Bash

**Project Example**: 
[selenoid-issue.zip](https://github.com/aerokube/selenoid/files/2345356/selenoid-issue.zip)

**STEPS:**
1. Start Selenoid service:
`docker-compose up`
Here seems to be everything is ok with the current configurations in docker-compose.yml. The directory `.data/` is mounted properly.

2. Run some tests using selenoid (execute test.py file)

**RESULT:**
Appears this error:
```
selenoid_1  | 2018/09/03 11:33:47 [0] [SERVICE_STARTUP_FAILED] [start video container: create video container: Error response from daemon: invalid volume specification: 'C:/Users/fedar/Documents/git/front-practice/selenoid-issue/.data/selenoid/videos:/data:rw']
```

**EXPLANATION:**
In the example, in the `OVERRIDE_VIDEO_OUTPUT_DIR` I'm using `${PWD}` for getting the current directory. I know I need to use the absolute path according to docker-compose documentation.
In Mac/Unix this command works properly and I get the current directory and everything is working OK. The problem is when I run this service on Windows. If you run the following command while is running the service:
```
docker-compose exec selenoid printenv
```
You will see is not resolving correctly the absolute path:
```
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=fd8b38501aff
TERM=xterm
OVERRIDE_VIDEO_OUTPUT_DIR=C:/Users/fedar/Documents/git/front-practice/selenoid-issue/.data/selenoid/videos
HOME=/root
```
The current value what is storing selenoid in `OVERRIDE_VIDEO_OUTPUT_DIR` is `C:/Users/fedar/Documents/git/front-practice/selenoid-issue/.data/selenoid/videos`
 but it should be `/c/Users/fedar/Documents/git/front-practice/selenoid-issue/.data/selenoid/videos` with the format `/c/...`, this is the way that docker needs to work properly.

If I run next command in GIT BASH, returns absolute path with the right format:
```
echo ${PWD}
```
![git bash](https://user-images.githubusercontent.com/1236418/44985230-785b0480-af77-11e8-9ad7-1136f5619ad8.png)
For that reason, I think is possible the selenoid container is not handling that path correctly.

It works if I hardcoded the absolute path like this:
```
    environment:
      - OVERRIDE_VIDEO_OUTPUT_DIR=/c/Users/fedar/Documents/git/front-practice/selenoid-issue/.data/selenoid/videos
```
```
$ docker-compose exec selenoid printenv
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=dd904aa2accc
TERM=xterm
OVERRIDE_VIDEO_OUTPUT_DIR=/c/Users/fedar/Documents/git/front-practice/selenoid-issue/.data/selenoid/videos
HOME=/root
```
Thank you.

**Comment by vania-pooh** [comment]
@fescobar seems to be `docker-compose` issue then. Do you consider this an issue in documentation?

**Comment by fescobar** [comment]
I don't know if it's an issue in the doc, but I haven't found an issue related to this. Let me see if I can create a docker-compose issue.

**Comment by fescobar** [comment]
@vania-pooh Could you tell me why are you using that VAR ENV? I mean you can store the videos in the mounted directory instead to use the value in the OVERRIDE_VIDEO_OUTPUT_DIR variable. Or internally do you use it for any other thing?

**Comment by vania-pooh** [comment]
@fescobar the problem is that Selenoid itself is running in Docker container, but it starts two containers when using video recorder:
1) Browser container
2) Video recorder container

This latter video recorder container is started with a volume where the video is actually saved, e.g.:
```
docker run -v /host/machine/dir:/data selenoid/video-recorder
```
But so far as Selenoid is also running inside container is does not know the real path `/host/machine/dir`. We are using this variable to send this knowledge to Selenoid. If you start Selenoid as a binary - no variable is needed.

**Comment by fescobar** [comment]
Hi @vania-pooh
I was working in a similar case that describes this issue. 
I think there is a workaround for this to work properly.
What happened if you consume the variable value from a temporary file instead to consume the value directly from **OVERRIDE_VIDEO_OUTPUT_DIR**.
I mean:
1. Read **OVERRIDE_VIDEO_OUTPUT_DIR** and save data in a file.
2. When you want to get this value, read that value from the file.
Then you could get that value with the proper format, in my case:
`
OVERRIDE_VIDEO_OUTPUT_DIR=/c/Users/fedar/Documents/git/front-practice/selenoid-issue/.data/selenoid/videos
`
Thank you,

**Comment by vania-pooh** [comment]
@fescobar I don't get what is the difference between using a file or an in-memory variable to store video output dir. This could also be a flag, I don't see any difference.

**Comment by fescobar** [comment]
Hi @vania-pooh 
I wasn't clear last time. I'm going to suggest a workaround for this, it's not the best solution but it could resolve this case. At least it worked in a container what I've created recently.

**PROBLEM RESUMEN**
- When I used ${PWD} for windows using GIT BASH or POWERSHELL. My absolute path is converted from `/c/directory/` to `C:/directory/`. As we know these tools are parsing that path.

docker-compose.yml
```
    environment:
      - OVERRIDE_VIDEO_OUTPUT_DIR=${PWD}/.data/selenoid/videos
```

**POSSIBLE SOLUTION**
I've created this sh `docker_parse_mounted_dir.sh`
```
#!/bin/bash
#Verify if OVERRIDE_VIDEO_OUTPUT_DIR contains ':' character to determine if it's for windows or not
if [[ $OVERRIDE_VIDEO_OUTPUT_DIR = *":"* ]]; then
  #Extract letter of disk, it could be any letter before character ':'
  DISK_LETTER=$(echo "$OVERRIDE_VIDEO_OUTPUT_DIR" | sed 's/\(.*\):\(.*\)/\1/')
  #Replace for example 'C:' by '/c/' 
  WIN_DIR_PARSED=$(echo "$OVERRIDE_VIDEO_OUTPUT_DIR" | sed 's/\(.*\):/\/\L'$DISK_LETTER'/')
  #Print path with format '/c/directory'
  echo $WIN_DIR_PARSED
else
  #Print normal path
  echo $OVERRIDE_VIDEO_OUTPUT_DIR
fi
```
- Basically, this script parses Windows path to format `/c/directory`. I'm not exporting the value for overriding the env variable "OVERRIDE_VIDEO_OUTPUT_DIR", because I understand that we can't do that in docker container at the moment.

If you are executing a script for running the selenoid container, you should do something like this:
```
DIR_MOUNTED=$(/any-directory-in-container/docker_parse_mounted_dir.sh)
```
And you could use the value with the right format and it works for all operating systems.
As I said before, it's not a clean solution. What do you think?

**Comment by madhank93** [comment]
It works for me. If I provide both the relative path and key inside the quotes.

```
environment:
      - "OVERRIDE_VIDEO_OUTPUT_DIR=$PWD/path/to/videos"
```

**Comment by facusantillo** [comment]
Any updates  / workaround on this?

**Comment by fescobar** [comment]
> Any updates / workaround on this?

Use Selenium Docker (Selenium Grid 4), which already includes VNC connections and videos.
